app/collections/show.jsx

```diff
@@ -85,41 +83,59 @@ const CollectionPage = React.createClass({
     const profileLink = `${baseLink}/users/${this.state.owner.login}`;
     const collectionsLinkMessageKey = `collectionPage.${baseType}Link`;
 
+    let roles;
+    if (this.props.roles.some(collectionRole => collectionRole.links.owner.id === this.props.user.id)) {
```

Comment:
This check doesn't seem necessary. You could eliminate the `roles` variable, set `userRole` to the returned value of the filter and L102 check for the length of `userRole`, then call a join on `userRole[0].roles`